Add FixUnmarshalIndividualSetValues option to DecoderOptions of dynamodb

AnatolyRugalev opened the pull request:
Fixes https://github.com/aws/aws-sdk-go-v2/issues/2895

AnatolyRugalev commented:
I am not sure how to add a changelog record. Feel free to take over this PR

lucix-aws commented:
JFYI we still plan to review/approve this, but it may have to wait until after the Thanksgiving week.

AnatolyRugalev commented on feature/dynamodb/attributevalue/decode.go:
```diff
@@ -243,6 +243,10 @@ type DecoderOptions struct {
+
+	// When enabled, the decoder will call Unmarshaler.UnmarshalDynamoDBAttributeValue
+	// for each individual set item instead of the whole set at once.
+	UnmarshalIndividualSetValues bool
```
Done

Madrigal commented:
We recently added instructions on how to generate a changelog entry for PRs. In the past, we had to do this on behalf of contributors, slowing down the review process. Can you go through [this document](https://github.com/aws/aws-sdk-go-v2/blob/main/CONTRIBUTING.md#changelog-documents) and generate a changelog entry for this PR?

lucix-aws commented:
Ping @AnatolyRugalev, please see previous comment about adding a changelog

AnatolyRugalev commented:
Sorry, I was on holidays and missed your message. I added the changlog as instructed and rebased the branch. Since this change is opt-in, I am marking my change as a feature instead of a bugfix. Let me know if you want me to change that
